connector/src/test/java/com/google/cloud/hive/bigquery/connector/integration/ImpersonationTests.java

```diff
@@ -0,0 +1,120 @@
+      return getErrorJsonFromHttpResponseException(
+          e.getCause(), stackTraceClassName, stackTraceMethodName);
+    }
+    return null;
+  }
+
+  @Test
+  public void testGlobalImpersonation() {
+    String sa = "abc@example.iam.gserviceaccount.com";
+    hive.setHiveConfValue("bq.impersonation.service.account", sa);
+    initHive();
+    Throwable exception =
+        assertThrows(
+            RuntimeException.class, () -> createManagedTable("abcd", HIVE_ALL_TYPES_TABLE_DDL));
+    JsonObject error =
+        getErrorJsonFromHttpResponseException(
+            exception, "com.google.auth.oauth2.ImpersonatedCredentials", "refreshAccessToken");
+    assertEquals(
+        error.getAsJsonObject("error").get("message").getAsString(),
+        "Not found; Gaia id not found for email " + sa);
+  }
+
+  @Test
+  public void testImpersonationForUser() {
+    String user = "bob";
+    String sa = "bob@example.iam.gserviceaccount.com";
+    hive.setHiveConfValue("bq.impersonation.service.account.for.user." + user, sa);
+    initHive();
+    UserGroupInformation ugi = UserGroupInformation.createRemoteUser(user);
+    ugi.doAs(
+        (java.security.PrivilegedAction<Void>)
+            () -> {
+              Throwable exception =
+                  assertThrows(
+                      RuntimeException.class,
+                      () -> createManagedTable("abcd", HIVE_ALL_TYPES_TABLE_DDL));
+              JsonObject error =
+                  getErrorJsonFromHttpResponseException(
+                      exception,
+                      "com.google.auth.oauth2.ImpersonatedCredentials",
+                      "refreshAccessToken");
+              assertEquals(
+                  error.getAsJsonObject("error").get("message").getAsString(),
+                  "Not found; Gaia id not found for email " + sa);
+              return null;
+            });
+  }
+
+  @Test
+  public void testImpersonationForGroup() {
+    String user = "bob";
+    String[] groups = new String[] {"datascience"};
+    String sa = "datascience-team@example.iam.gserviceaccount.com";
+    hive.setHiveConfValue("bq.impersonation.service.account.for.group." + "datascience", sa);
+    initHive();
+    UserGroupInformation ugi = UserGroupInformation.createUserForTesting(user, groups);
+    ugi.doAs(
+        (java.security.PrivilegedAction<Void>)
+            () -> {
+              Throwable exception =
+                  assertThrows(
+                      RuntimeException.class,
+                      () -> createManagedTable("abcd", HIVE_ALL_TYPES_TABLE_DDL));
+              JsonObject error =
+                  getErrorJsonFromHttpResponseException(
+                      exception,
+                      "com.google.auth.oauth2.ImpersonatedCredentials",
+                      "refreshAccessToken");
+              assertEquals(
+                  error.getAsJsonObject("error").get("message").getAsString(),
+                  "Not found; Gaia id not found for email " + sa);
```

Comment:
Can we show group info here?

Reply:
this error is at config initiation time to check the service account, the error message contains the sa, the linux user info that actually runs the job is not available here.